build.gradle

```diff
@@ -51,6 +53,31 @@ if(JavaVersion.current() != JavaVersion.VERSION_1_8) {
     throw new GradleException("This project requires Java 8, but it's running on " + JavaVersion.current())
 }
 
+static String getVersionHash(String buildScriptContent) {
+    String versionLine = buildScriptContent.find("^//version: [a-z0-9]*")
+    if(versionLine != null) {
+        return versionLine.split(": ").last()
+    }
+    return ""
+}
+
+static def checkForUpdate() {
+    Thread.start {
+        Map parameters = ["connectTimeout": 2000, "readTimeout": 2000]
+        
+        String currentBuildScript = new File("build.gradle").getText()
+        String currentBuildScriptHash = getVersionHash(currentBuildScript)
+        
+        String availableBuildScript = new URL("https://raw.githubusercontent.com/SinTh0r4s/ExampleMod1.7.10/main/build.gradle").newInputStream(parameters).getText()
```

Comment:
in my tests execution hanged in this part, @SinTh0r4s any ideas why?

Reply:
It works perfectly fine for me. Keep in mind, that `main` does not yet have any hash in `build.gradle`. You have to test against this branch, which works fine

Reply:
I've added test prints to console and I should've received at least an empty string, but there was none, as if it didn't manage to get there.